pyccel/codegen/printing/ccode.py

```diff
@@ -1963,6 +2010,31 @@ def _print_FunctionCall(self, expr):
         else:
             return call_code
 
+    def init_stc_container(self, expr, container_var):
+        """
+        Generate the initialization of an STC container in C.
+
+        This method generates and prints the C code for initializing a container using the STC `init()` method.
+
+        Parameters
+        ----------
+        expr : TypedAstNode
+            The object representing the container being printed (e.g., PythonList, PythonSet).
+    
+        container_var : str
+            The variable name to which the container is being assigned.
+        
+        Returns
+        -------
+        str
+            The generated C code for the container initialization.
+        """
+
+        dtype = self.find_in_dtype_registry(container_var.lhs.class_type)
+        keyraw = '{' + ', '.join([self._print(a) for a in expr.args]) + '}'
```

Comment:
```suggestion
        keyraw = '{' + ', '.join(self._print(a) for a in expr.args) + '}'
```
`join` takes an iterator. This is usually better as it avoids an unnecessary memory allocation